content-api.apib

```diff
@@ -0,0 +1,483 @@
+
+In Superdesk, a News Item is a generic container for any type of news content.
+Photos, text, audio, video and graphics are all represented as News Items.
+
+## News Item [/items/{_id}]
+
+
+### Fetch a single News Item [GET]
+
+This method is used to retrieve a particular item as a NewsML-G2 News Item
+document in its ninjs (News in JSON) representation. To invoke this method, you
+must know the item Id of the News Item you want to retrieve.
+
++ Parameters
+    + _id (required, string, `1`) ... Identifier of the News Item you want to retrieve.
+
++ Response 200 (application/json)
+
+            {
+                "_etag": "abc",
+                "_links": {
+                    "self": {
+                        "href": "/media/1"
+                    }
+                },
+                "abstract": "abstract needed...",
+                "byline": "John Doe",
+                "copyrightholder": "Sourcefabric z.u.",
+                "copyrightnotice": "(c) Copyright Sourcefabric z.u. 2015",
+                "description_text": "Flower picture",
+                "filemeta": {
+                    "Manufacturer": "CANON", "Orientation": "landscape"
+                },
+                "firstcreated": "2013-07-03T10:11:12Z",
+                "headline": "No news today",
+                "language": "en",
+                "mimetype": "image/jpeg",
+                "provider": "My Company LTD",
+                "pubstatus": "usable",
+                "renditions": {
+                    "original": {
+                        "height": 3462,
+                        "href": "http://farm4.staticflickr.com/3665/9203816834_9f62964627_o_d.jpg",
+                        "mimetype": "image/jpeg",
+                        "width": 5198
+                    },
+                    "thumbnail": {
+                        "height": 150,
+                        "href": "http://farm4.staticflickr.com/3665/9203816834_3329fac058_q_d.jpg",
+                        "mimetype": "image/jpeg",
+                        "width": 150
+                    }
+                },
+                "type": "picture",
+                "urgency": 4,
+                "uri": "tag:mycomp-ltd.com,0000:newsml_LM1EA5F0UUD01",
+                "usageterms": "You are not allowed.",
+                "version": "1",
+                "versioncreated": "2013-07-03T10:11:12Z"
+            }
+
+## News Item Collection [/items?q={q}&type={type}&date_range={date_range}&fields={fields}&limit={limit}]
+
+### List News Items [GET]
+
+This method is used to retrieve a list of News Items.
+
++ Parameters
+    + q (optional, object) ... Elasticsearch query
+    + type (optional, string) ... Specifies the content type for the news items to be
+      retrieved. Valid content types are: text, image, video, audio and composite. Default
+      value: All available media types will be retrieved.
+    + date_range (optional, date interval) ... The date range for the news items to be
```

Comment:
1.) I prefer flat parameters as well, e.g. `start_date` and `end_date`, especially because one of the pair can be omitted in queries like _"give me the items created after XYZ"_.

2.) We need to make sure that it's perfectly clear whether the date range boundaries are inclusive or exclusive (a common source of confusion). Preferably by finding even more descriptive, yet still nice, parameter names (e.g. _newer_than_)

3.) Since the parameters' data type will most likely be date**time** (to be decided...), their names should reflect that to avoid any confusion.